test/audio/PlaybackQueue-test.ts

```diff
@@ -0,0 +1,254 @@
+                needToBeTested = false;
+            }
+        });
+    });
+
+    it('a non-voice message stop the queue', async () => {
+        const roomIdTest = '!room1:server';
+        const ev0 = mkEvent({
+            event: true,
+            type: "m.room.message",
+            user: "@user1:server",
+            room: roomIdTest,
+            content: {
+                "org.matrix.msc2516.voice": {},
+            },
+        });
+
+        const ev1 = mkEvent({
+            event: true,
+            type: "m.room.message",
+            user: "@user1:server",
+            room: roomIdTest,
+            content: {},
+        });
+
+        const ev2 = mkEvent({
+            event: true,
+            type: "m.room.message",
+            user: "@user1:server",
+            room: roomIdTest,
+            content: {
+                "org.matrix.msc2516.voice": {},
+            },
+        });
+
+        updateCli([ev0, ev1, ev2], roomIdTest);
+
+        const playback0 = PlaybackManager.instance.createPlaybackInstance(buffer, waveform);
+        const playback1 = PlaybackManager.instance.createPlaybackInstance(buffer, waveform);
+        const PlaybackQueueInstance = PlaybackQueue.forRoom(ev0.getRoomId());
+        PlaybackQueueInstance.unsortedEnqueue(ev0, playback0);
+        await playback0.prepare();
+        PlaybackQueueInstance.unsortedEnqueue(ev2, playback1);
+        await playback1.prepare();
+
+        await playback0.toggle();
+        await playback0.stop();
+
+        playback1.on(UPDATE_EVENT, (state) => {
+            expect(state).toBe(PlaybackState.Stopped);
+        });
+    });
+
+    it('paused voice messages are not auto played after the queue', async () => {
+        const roomIdTest = '!room2:server';
+        const ev0 = mkEvent({
+            event: true,
+            type: "m.room.message",
+            user: "@user1:server",
+            room: roomIdTest,
+            content: {
+                "org.matrix.msc2516.voice": {},
+            },
+        });
+
+        const ev1 = mkEvent({
+            event: true,
+            type: "m.room.message",
+            user: "@user1:server",
+            room: roomIdTest,
+            content: {
+                "org.matrix.msc2516.voice": {},
+            },
+        });
+
+        updateCli([ev0, ev1], roomIdTest);
+
+        const playback0 = PlaybackManager.instance.createPlaybackInstance(buffer, waveform);
+        const playback1 = PlaybackManager.instance.createPlaybackInstance(buffer, waveform);
+        const PlaybackQueueInstance = PlaybackQueue.forRoom(ev0.getRoomId());
+        PlaybackQueueInstance.unsortedEnqueue(ev0, playback0);
+        await playback0.prepare();
+        PlaybackQueueInstance.unsortedEnqueue(ev1, playback1);
+        await playback1.prepare();
+
+        await playback0.toggle();
+        await playback0.pause();
+
+        await playback1.toggle();
+        await playback1.stop();
```

Comment:
Could we call `destroy()` instead, usage of `stop()` seems confusing

Reply:
we cannot as `destroy()` calls `stop()` and `removeAllListeners()` the event is not received by `PlaybackQueue.ts` so the wrongful behavior cannot be tested

Reply:
same thing for the other cases

Reply:
Ah, thanks for the explanation!